apps/web/pages/apps/installation/[[...step]].tsx

```diff
@@ -155,47 +182,46 @@ const OnboardingPage = ({
   });
 
   const handleSelectAccount = async (teamId?: number) => {
-    try {
-      setIsSelectingAccount(true);
-      if (appMetadata.isOAuth) {
-        const state = JSON.stringify({
-          appOnboardingRedirectUrl: getAppOnboardingRedirectUrl(appMetadata.slug, teamId),
-          teamId,
-        });
-
-        const res = await fetch(
-          `/api/integrations/${
-            appMetadata.slug == "stripe" ? "stripepayment" : appMetadata.slug
-          }/add?state=${state}`,
-          {
-            method: "GET",
-            headers: {
-              "Content-Type": "application/json",
-            },
+    mutation.mutate(
+      {
+        isOmniInstall: true,
+        type: appMetadata.type,
+        variant: appMetadata.variant,
+        slug: appMetadata.slug,
+        ...(teamId && { teamId }),
+        // for oAuth apps
+        ...(showEventTypesStep && {
+          returnTo:
+            WEBAPP_URL +
+            getAppOnboardingUrl({
+              slug: appMetadata.slug,
+              teamId,
+              step: AppOnboardingSteps.EVENT_TYPES_STEP,
+            }),
+        }),
+      },
+      {
+        onSuccess: (data) => {
+          if (data?.setupPending) return;
+          if (showEventTypesStep) {
+            // for non-oAuth apps
+            router.push(
+              getAppOnboardingUrl({
+                slug: appMetadata.slug,
+                step: AppOnboardingSteps.EVENT_TYPES_STEP,
+                teamId,
+              })
+            );
```

Comment:
Do you think it's worth adding this redirect logic in the installation process to the `useAddAppMutation` hook? We're already redirecting to other pages there.

Reply:
We have a follow-up PR coming up 😅, that will improve app installations in organizations, -- will refactor useAddAppMutation in that.

Reply:
pushed a refactor in 

https://github.com/calcom/cal.com/pull/15193/commits/b2f478cdb68ab0c5caf30c581e10547548bafbbd and https://github.com/calcom/cal.com/pull/15193/commits/69378039ec5df39cd7e74d39bf4c3fd148ac537b

@joeauyeung @zomars